test/micro/org/openjdk/bench/vm/runtime/NMTBenchmark_wb.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.vm.runtime;
+
+import org.openjdk.bench.vm.lang.Throw;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.BenchmarkParams;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+import java.lang.*;
+import java.util.*;
+
+import jdk.internal.misc.Unsafe;
+import jdk.test.whitebox.WhiteBox;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+@Warmup(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+public abstract class NMTBenchmark_wb {
+
+    @Param({"2", "4", "8", "16"})
+    public int THREADS;
+
+    @Param({"16", "32", "64"})
+    public int REGIONS;
+
+    @Param({"4", "8", "16", "32", "64"})
+    public int SUB_REGIONS;
+
+    private static final int PageSize = 1024 * 4;
+
+    // Need to wrap WhiteBox in a holder class so that it doesn't get initialized on the host VM (which won't
+    // have WB enabled.
+    private static class WhiteBoxHolder {
+
+        private static final WhiteBox WB;
+
+        static {
+            System.out.println("Current user dir: " + System.getProperty("user.dir"));
+            WhiteBox wb = null;
+            try {
+                wb = WhiteBox.getWhiteBox();
+            } catch (Throwable t) {
+                System.out.println("WhiteBox initialization failure: " + t);
+                t.printStackTrace();
+                // Explicit exit to avoid initialization loops.
+                System.exit(17);
+            }
+            WB = wb;
+        }
+    }
+
+    private static long reserve (long size           ) { return WhiteBoxHolder.WB.NMTReserveMemory (size                           ); }
+    private static void commit  (long base, int pno  ) {        WhiteBoxHolder.WB.NMTCommitMemory  (base + pno * PageSize, PageSize); }
+    private static void uncommit(long base, int pno  ) {        WhiteBoxHolder.WB.NMTUncommitMemory(base + pno * PageSize, PageSize); }
+    private static void release (long base, long size) {        WhiteBoxHolder.WB.NMTReleaseMemory (base                 , size    ); }
+
+    public static void doAllMemoryOps(int nR, int region_count) {
+        long region_size = region_count * PageSize;
+        long[] base_array = new long[nR];
+        for (int i = 0; i < nR; i++)
+          base_array[i] = reserve(region_size);
+
+        for (int R = 0; R < nR; R++) {
```

Comment:
The capitol `R` doesn't look quite right to me here, maybe use `reservation` or even a simple plain `r` would look better, imho.

Reply:
Done.